### PR TITLE
fix: Enhance chat response handling for v1 and v2 endpoints

### PR DESCRIPTION
**Pull request by mraman-2U**
This pull request introduces support for handling both V1 and V2 response formats in the XPert Chat API, improves timezone handling in the codebase and tests, and updates the version to `4.10.8`. The most significant changes include the addition of logic to process V2 API responses, updates to ensure consistent timezone usage, and the addition of comprehensive test cases for the new functionality.

### Support for XPert Chat API V2:
* Updated `_get_next_message` in `learning_assistant/views.py` to handle V2 endpoint responses, which return an array of messages, while maintaining support for V1 responses.
* Added a new toggle import `v2_endpoint_enabled` in `learning_assistant/views.py` to enable conditional handling of V2 responses.

### Timezone Handling Improvements:
* Introduced `USE_TZ` and `TIME_ZONE` settings in `test_settings.py` to ensure consistent timezone handling during tests.
* Replaced all instances of `datetime.now()` with `timezone.now()` across test files (`tests/test_api.py`, `tests/test_views.py`) to use timezone-aware timestamps. [[1]](diffhunk://#diff-3451cc474d5698222168f66a293339a6bd296ee2c56f56409866bbae99917a79L571-R584) [[2]](diffhunk://#diff-bbc49e4202aabab5bb86711f4824030832a25188901ab84ab72e7e550b66603cL244-R246) and others)
* Adjusted test assertions to account for timezone-aware timestamps, including handling differences between ISO 8601 formats.

### Testing Enhancements:
* Added test cases in `tests/test_utils.py` and `tests/test_views.py` to validate both V1 and V2 response formats, including edge cases for malformed or unexpected responses. [[1]](diffhunk://#diff-33c13e0b177bacd2f02e29bcb8aea5b49e7ce34901fd8f41fefb65defba1bd33R151-R175) [[2]](diffhunk://#diff-bbc49e4202aabab5bb86711f4824030832a25188901ab84ab72e7e550b66603cR349-R498)
* Updated existing tests to ensure compatibility with the new timezone-aware implementation. [[1]](diffhunk://#diff-3451cc474d5698222168f66a293339a6bd296ee2c56f56409866bbae99917a79L611-R625) [[2]](diffhunk://#diff-bbc49e4202aabab5bb86711f4824030832a25188901ab84ab72e7e550b66603cL429-R589)

### Documentation and Versioning:
* Updated `CHANGELOG.rst` to reflect the new version `4.10.8` and the added support for XPert Chat API V2.
* Incremented the version in `learning_assistant/__init__.py` to `4.10.8`.

### CODE REVIEW

File: learning_assistant/views.py [hunk 1]
```diff
@@ -115,7 +115,22 @@ def _get_next_message(self, request, courserun_key, course_run_id):
         status_code, message = get_chat_response(prompt_template, message_list)
 
         if chat_history_enabled(courserun_key):
-            save_chat_message(courserun_key, user_id, LearningAssistantMessage.ASSISTANT_ROLE, message['content'])
+            if v2_endpoint_enabled() and isinstance(message, list):
```
Comment (michaelroytman) [inline learning_assistant/views.py:118]: We don't have to worry about tool calls, right?

Comment (michaelroytman) [inline learning_assistant/views.py:118]: Maybe we could stick this logic in a utility function?

File: learning_assistant/views.py [hunk 2]
```diff
@@ -115,7 +115,22 @@ def _get_next_message(self, request, courserun_key, course_run_id):
...
+                else:
+                    content = ''  # Fallback for empty list
+            elif isinstance(message, dict) and 'content' in message:
+                # For v1 endpoint, message is a dict with content key
+                content = message['content']
+            else:
+                # Fallback for other formats (e.g., error strings)
+                content = str(message)
+
+            save_chat_message(courserun_key, user_id, LearningAssistantMessage.ASSISTANT_ROLE, content)
```
Comment (michaelroytman) [inline learning_assistant/views.py:133]: Do we have to also consider the call to `save_chat_message` on line 88?

**Review by michaelroytman — COMMENTED** [review]
I left a few clarifying questions.

File: learning_assistant/views.py (hunk 1 above)
Comment (mraman-2U) [reply learning_assistant/views.py:118]: As no tools have been used in our learning assistant, we no need to worry about tool calls

File: learning_assistant/views.py (hunk 2 above)
Comment (mraman-2U) [reply learning_assistant/views.py:133]: On line 88, the input sent to the learning assistant being recorded. As this is not the reponse from Chat api, it is not required to handle here.

File: learning_assistant/views.py (hunk 1 above)
Comment (mraman-2U) [reply learning_assistant/views.py]: Refactored the message extraction logic as a utility function

**Review by michaelroytman — APPROVED** [review]
